Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,928 @@
+  private static _requiredSelectionAny: number = 0;
+  private static _requiredSelectionExactlyOne: number = 1;
+  private static _requiredSelectionOneOrMore: number = 2;
+  private static _itemClickHandler: Nullable<(e: JQueryEventObject) => boolean> = null;
+  private static _itemClicked: boolean = false;
+  private static _repositionInterval: number = 200;
+  private static _repositionTimer: Nullable<number> = null;
+  private static _statusPopupDelay: number = 200;
+  private static _statusPopupRepositionTimer: Nullable<number> = null;
+  private static _blurTimer: Nullable<number> = null;
+  private static _updateFocus: boolean = true;
+
+  public static AddMenuInfo(menuInfo: DropDownMenu_MenuInfo): void
+  {
+    DropDownMenu._menuInfos[menuInfo.ID] = menuInfo;
+  }
+
+  public static BindOpenEvent (openTarget: JQuery, menuID: string, eventType: string, getSelectionCount: Nullable<() => number>, moveToMousePosition: boolean): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery ('openTarget', openTarget);
+    ArgumentUtility.CheckNotNullAndTypeIsString ('menuID', menuID);
+    ArgumentUtility.CheckNotNullAndTypeIsString ('eventType', eventType);
+    ArgumentUtility.CheckNotNullAndTypeIsBoolean ('moveToMousePosition', moveToMousePosition);
+
+    var menuButton = openTarget[0].tagName.toLowerCase() === 'a' ? openTarget : openTarget.find ('a[href]:last');
+    menuButton.attr('aria-haspopup', 'menu');
+    menuButton.attr('aria-expanded', 'false');
+
+    openTarget.bind(
+        eventType,
+        function (evt: JQueryMouseEventObject)
+        {
+          menuButton.focus();
+          DropDownMenu.OnClick (openTarget, menuID, getSelectionCount, moveToMousePosition ? evt : null);
+        });
+
+    if (!moveToMousePosition)
+    {
+      menuButton
+          .bind ("focus", function (evt) { openTarget.find ('.' + DropDownMenu._buttonClassName).addClass (DropDownMenu._focusClassName); })
+          .bind("blur", function (evt) { openTarget.find('.' + DropDownMenu._buttonClassName).removeClass (DropDownMenu._focusClassName); });
+
+      var allButMenuButton = openTarget.find ('a[href]').not(menuButton);
+      allButMenuButton
+          .bind ("mouseover", function (evt) { openTarget.find ('.' + DropDownMenu._buttonClassName).addClass (DropDownMenu._nestedHoverClassName); })
+          .bind ("mouseout", function (evt) { openTarget.find ('.' + DropDownMenu._buttonClassName).removeClass (DropDownMenu._nestedHoverClassName); });
+    }
+  }
+
+  public static LoadFilteredMenuItems(
+      itemInfos: DropDownMenu_ItemInfo[],
+      loadMenuItemStatus: Nullable<(s: (i: DropDownMenu_ItemInfo[]) => void, e: () => void) => void>,
+      onSuccess: (i: DropDownMenu_ItemInfo[]) => void,
+      onError: () => void): void
+  {
+    if (loadMenuItemStatus !== null)
+    {
+      const currentLoadOperationID = new Object();
+      const onSuccessForLoadMenuItemStatus = function (itemStatusArray: DropDownMenu_ItemInfo[])
```

Comment:
We should explicitly add the type annotation to handlers, because there can be similar ducking problems when the function type is changed.